Anagram.java

```diff
@@ -28,22 +28,63 @@ public static void main(String args[]) {
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
+		String new1 = preProcess(str1);
+		String new2 = preProcess(str2);
+		char c;
+		for (int i = 0; i < new1.length(); i++) {
+			String cut = "";
+			int indexStr2;
+			c = new1.charAt(i);
+			indexStr2 = new2.indexOf(c);
+
+			if(indexStr2 == -1) {
+				return false;
+			}
+			for (int j = 0; j < new2.length(); j++) {
+				if (j != indexStr2) {
+					cut += new2.charAt(j);
+				}
+			}
+			new2 = cut;
```

Comment:
why not
`new2 = new2.substring(0, indexStr2) + new2.substring(indexStr2 + 1)`
if you are worried about case `indexStr2 = new2.length() - 1` you can wrap it with `if` and in the case use only `new2 = new2.substring(0, indexStr2)`. alternatively: 
```
String updatedNew2 = new2.substring(0, indexStr2);
if (indexStr2 != new2.length() - 1) {
   updatedNew2 +=  new2.substring(indexStr2 + 1);
}
new2 = updatedNew2;
```